Updated welcome screen to a newer style

heathermh opened the pull request:
Updated layout to single screen, updated to be in line with apple standard designs. Friendly for dark mode and dynamic text. Includes accessibility improvements to work better with VoiceOver and Voice Control.

Partially resolves https://github.com/5calls/ios/issues/331


![](https://user-images.githubusercontent.com/1638675/281218199-504a9ab0-f915-4c93-8a75-ede39253c363.png)

![](https://user-images.githubusercontent.com/1638675/281218511-ac534b30-b8a2-4051-ab34-0abe78151fff.png)

nickoneill commented on FiveCalls/FiveCalls/Localizable.strings:
```diff
@@ -132,6 +132,7 @@
 "subheading-message"                  = "You're making a difference! Calling is the most effective way to influence your representatives, and here's what you've achieved:";
 
 // welcome
+"welcome-logo-name"                   = "Five Calls";
```
Is this used somewhere?

heathermh replied:
Nope, pulled that bit. Cleaned up now.

nickoneill commented on FiveCalls/FiveCalls/Welcome.swift:
```diff
@@ -9,112 +9,127 @@
+                        }
+                        .accessibilityElement(children: .combine)
+                    } icon: {
+                        Image(systemName: "person.2")
```
Can we align this icon so the centers of all the icons are centered? This last one seems a little to the right of the others

heathermh replied:
I'll see what I can do. Are the symbol choices OK so far?

nickoneill replied:
Yep, they're great. Maybe bump the size too? they're a bit hard to see right now.

nickoneill commented on FiveCalls/FiveCalls/Welcome.swift:
```diff
@@ -9,112 +9,127 @@
+                                .fontWeight(.heavy)
+                                .opacity(subMessageOpacity)
+                        }
+                        .accessibilityElement(children: .combine)
```
I haven't used any accessibility components in SwiftUI yet so I might be misunderstanding here, but in reading the docs for this it sounds like this would be used to combine multiple views into a single element that would used when iterating through accessible items on the page. What does this do if there's only one view within the VStack?

heathermh replied:
Yep, we can leave this off.

nickoneill commented on FiveCalls/FiveCalls/Welcome.swift:
```diff
@@ -9,112 +9,127 @@
+                        Image(systemName: "person.2")
+                            .opacity(subMessageOpacity)
+                            .symbolRenderingMode(.palette)
+                            .foregroundStyle(.fivecallsGreen, .fivecallsDarkBlue)
```
Can we use `.fivecallsRed` as the secondary color like above? We don't really use green anywhere in the app.

heathermh commented:
Adjusted alignment of symbols, made symbols bigger, and adjusted the colors a bit. We could add a little red in the spend 5 minutes symbol, if you like. I removed the period after "Spend 5 minutes, make 5 calls" so it's more consistent with other content.

Still ready to go for dark mode, dynamic text, accessibility support (VoiceOver, Voice Control). One small issue I see is that on the iPhone SE, the continue button is a little cut off. Not sure if it's worth adjusting for? I'll include a screen shot of it below.
<img width="316" alt="image" src="https://github.com/5calls/ios/assets/1638675/d11c863f-3c18-4b63-b6d0-95461a9e25f2">
<img width="687" alt="image" src="https://github.com/5calls/ios/assets/1638675/4272ffc8-5eaa-4ea0-8d12-5d04ad388b5c">
<img width="317" alt="image" src="https://github.com/5calls/ios/assets/1638675/a1c70920-f19e-47ed-b6f6-3da3e3fdb035">
<img width="317" alt="image" src="https://github.com/5calls/ios/assets/1638675/a9ae6289-13f5-4161-808a-4455066cdab9">

nickoneill commented on FiveCalls/FiveCalls/LocationHeader.swift:
```diff
@@ -53,6 +53,9 @@ struct LocationHeader: View {
                 .padding(.leading, 7)
         }
+        .accessibilityElement(children: .ignore)
+        .accessibilityLabel(Text("Your location is: \(location!.locationDisplay)"))
```
this should be in the localizable strings file

nickoneill commented on FiveCalls/FiveCalls/LocationHeader.swift:
```diff
@@ -71,6 +74,9 @@ struct LocationHeader: View {
                 .padding(.leading, 7)
         }
+        .accessibilityElement(children: .ignore)
+        .accessibilityLabel(Text("Set your location"))
```
add this to localizable as well

nickoneill commented on FiveCalls/FiveCalls/Welcome.swift:
```diff
@@ -9,112 +9,107 @@
+                        .opacity(subMessageOpacity)
+                        .gridColumnAlignment(.leading)
+                }
+                Spacer()
```
Do we need two spacers here?

heathermh replied:
For the welcome view to accommodate dynamic text, I wrapped it in a ScrollView. That changes how spacers work, and two spacers gives us good space / drops the button almost to the bottom with default text sizes and a modern phone. This is also why the button gets cut off a bit on the small phone.

nickoneill commented on FiveCalls/FiveCalls/Localizable.strings:
```diff
@@ -132,12 +133,12 @@
+"welcome-section-2-title"                = "Spend 5 minutes, make 5 calls";
+"welcome-section-2-message"              = "Calling is the most effective way to influence your representative.";
+"welcome-section-3-calls"                = "Together we've made\n%@ calls";
+"welcome-button-title"                   = "Continue";
```
I think "Get Started" makes more sense here than "Continue" since we only have one welcome screen and you get started right away

nickoneill reviewed (approved):
Looks great, a few comments to adjust and then feel free to merge.

In the future it would be nice to see unrelated changes like the location header accessibility changes in another small PR. it's easier to review smaller changesets and I certainly don't mind having more PRs to review
